src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceSync.kt

```diff
@@ -115,33 +108,13 @@ class InstanceSync(aem: AemExtension, instance: Instance) : InstanceHttpClient(a
         return response
     }
 
-    fun downloadPackage(remotePath: String, targetFile: File, retry: Retry = Retry.none()) {
-        lateinit var exception: FileException
-
-        for (i in 0..retry.times) {
-            try {
-                downloadPackageOnce(remotePath, targetFile)
-                return
-            } catch (e: FileException) {
-                exception = e
-
-                if (i < retry.times) {
-                    aem.logger.warn("Cannot download package $remotePath from $instance.")
-                    aem.logger.debug("Download error", e)
-
-                    val delay = retry.delay(i + 1)
-                    val countdown = ProgressCountdown(project, delay)
-
-                    aem.logger.lifecycle("Retrying download (${i + 1}/${retry.times}) after delay: ${Formats.duration(delay)}")
-                    countdown.run()
-                }
-            }
+    fun downloadPackage(remotePath: String, targetFile: File, retry: Retry = aem.retry()) {
+        return retry.launch<Unit, InstanceException>("download package") {
```

Comment:
launch is too generic in my opinion, consider replacing with `retry.launchAndStopOnException`

Reply:
I was think about run apply etc but these are kotlin top level high order functions / names should be avoided. Yours is too descriptive in my opinion 😉